web/auth/client.go

```diff
@@ -0,0 +1,154 @@
+// Protocol.
+type Client struct {
+	CouchID  string `json:"_id,omitempty"`  // Generated by CouchDB
+	CouchRev string `json:"_rev,omitempty"` // Generated by CouchDB
+
+	ClientID          string `json:"client_id,omitempty"`       // Same as CouchID
+	ClientSecret      string `json:"client_secret"`             // Generated by the server
+	SecretExpiresAt   int    `json:"client_secret_expires_at"`  // Forced by the server to 0 (no expiration)
+	RegistrationToken string `json:"registration_access_token"` // Generated by the server
+
+	RedirectURIs    []string `json:"redirect_uris"`              // Declared by the client (mandatory)
+	GrantTypes      []string `json:"grant_types"`                // Forced by the server to ["authorization_code", "refresh_token"]
+	ResponseTypes   []string `json:"response_types"`             // Forced by the server to ["code"]
+	ClientName      string   `json:"client_name"`                // Declared by the client (mandatory)
+	ClientKind      string   `json:"client_kind,omitempty"`      // Declared by the client (optional, can be "desktop", "mobile", "browser", etc.)
+	ClientURI       string   `json:"client_uri,omitempty"`       // Declared by the client (optional)
+	LogoURI         string   `json:"logo_uri,omitempty"`         // Declared by the client (optional)
+	PolicyURI       string   `json:"policy_uri,omitempty"`       // Declared by the client (optional)
+	SoftwareID      string   `json:"software_id"`                // Declared by the client (mandatory)
+	SoftwareVersion string   `json:"software_version,omitempty"` // Declared by the client (optional)
+}
+
+// ID returns the client qualified identifier
+func (c *Client) ID() string { return c.CouchID }
+
+// Rev returns the client revision
+func (c *Client) Rev() string { return c.CouchRev }
+
+// DocType returns the client document type
+func (c *Client) DocType() string { return ClientDocType }
+
+// SetID changes the client qualified identifier
+func (c *Client) SetID(id string) { c.CouchID = id }
+
+// SetRev changes the client revision
+func (c *Client) SetRev(rev string) { c.CouchRev = rev }
+
+// ClientRegistrationError is a Client Registration Error Response, as described
+// in the Client Dynamic Registration Protocol
+// See https://tools.ietf.org/html/rfc7591#section-3.2.2 for errors
+type ClientRegistrationError struct {
+	Code        int    `json:"-"`
+	Error       string `json:"error"`
+	Description string `json:"error_description,omitempty"`
+}
+
+// Create is a function that sets some fields, and then save it in Couch.
+func (c *Client) Create(i *instance.Instance) *ClientRegistrationError {
+	if len(c.RedirectURIs) == 0 {
+		return &ClientRegistrationError{
+			Code:        http.StatusBadRequest,
+			Error:       "invalid_redirect_uri",
+			Description: "redirect_uris is mandatory",
+		}
+	}
+	for _, redirectURI := range c.RedirectURIs {
+		u, err := url.Parse(redirectURI)
+		if err != nil ||
+			(u.Scheme != "http" && u.Scheme != "https") ||
```

Comment:
For desktop, it will be something like `http://localhost:3000/oauth/callback`, but for mobile, you are right! I'm going to fix this